### PR TITLE
Fix "ReferenceError: url is not defined" in challenge

### LINKED ISSUE #14002: "ReferenceError: url is not defined" in challenge "Post Data with the JavaScript XMLHttpRequest Method"
Challenge [post-data-with-the-javascript-xmlhttprequest-method](https://beta.freecodecamp.com/en/challenges/json-apis-and-ajax/post-data-with-the-javascript-xmlhttprequest-method) has an issue.
User Agent is: <code>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36</code>.

The code below actually yields a `ReferenceError: url is not defined` on the console. Yet, all tests are passing. Where is the variable `url` supposed to come from? Apologies, if this is work in progress, but I thought I'd point it out. Feel free to rope me in for fixing this—dev environment is all set up 😀 

```html


<script>
  document.addEventListener('DOMContentLoaded',function(){
    document.getElementById('sendMessage').onclick=function(){
      var userName=document.getElementById('name').value;
      // Add your code below this line
      req=new XMLHttpRequest();
      try {
        req.open("POST",url,true);
      }
      catch (e) {
        console.log(e);
      }
      req.setRequestHeader('Content-Type','text/plain');
      req.onreadystatechange=function(){
        if(req.readyState==4 && req.status==200){
          document.getElementsByClassName('message')[0].innerHTML=req.responseText;
        }
      };
      req.send(userName);
      
      // Add your code above this line
    };
  });
</script>
<style>
  body {
    text-align: center;
    font-family: "Helvetica", sans-serif;
  }
  h1 {
    font-size: 2em;
    font-weight: bold;
  }
  .box {
    border-radius: 5px;
    background-color: #eee;
    padding: 20px 5px;
  }
  button {
    color: white;
    background-color: #4791d0;
    border-radius: 5px;
    border: 1px solid #4791d0;
    padding: 5px 10px 8px 10px;
  }
  button:hover {
    background-color: #0F5897;
    border: 1px solid #0F5897;
  }
</style>
<h1>Cat Friends</h1> 
<p class="message box">
  Reply from Server will be here
</p>
<p>
  <label for="name">Your name:
    <input type="text" id="name"/>
  </label>
  <button id="sendMessage">
    Send Message
  </button>
</p>

```

### PR DESCRIPTION
**Pull request by alfredojry**
I fix the challenge because issue #14002 
- The `url` variable is set to external resource `https://jsonplaceholder.typicode.com/posts`
- I replaced `.textContent` instead of `.innerHTML` (better practice)
- Checking for HTTP status of 201 instead of 200
All these suggestions by @HarplingeTom and "approved" by  @QuincyLarson. See the issue comments.
I test the code locally in my computer and this works :+1: 
I modified and verify the tests according the new assessment.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #14002

### CODE REVIEW

File: curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/post-data-with-the-javascript-xmlhttprequest-method.english.md [hunk 1]
```diff
@@ -8,11 +8,11 @@ challengeType: 6
...
 JavaScript's <code>XMLHttpRequest</code> method is also used to post data to a server. Here's an example:
-<blockquote>req=new XMLHttpRequest();<br>req.open("POST",url,true);<br>req.setRequestHeader('Content-Type','text/plain');<br>req.onreadystatechange=function(){<br>&nbsp;&nbsp;if(req.readyState==4 && req.status==200){<br>&nbsp;&nbsp;&nbsp;&nbsp;document.getElementsByClassName('message')[0].innerHTML=req.responseText;<br>&nbsp;&nbsp;}<br>};<br>req.send(userName);</blockquote>
+<blockquote>xhr = new XMLHttpRequest();<br>xhr.open('POST', url, true);<br>xhr.setRequestHeader('Content-Type', 'application/json; charset=UTF-8');<br>xhr.onreadystatechange = function () {<br>&nbsp;&nbsp;if (xhr.readyState === 4 && xhr.status === 201){<br>&nbsp;&nbsp;&nbsp;&nbsp;const serverResponse = JSON.parse(xhr.response);<br>&nbsp;&nbsp;&nbsp;&nbsp;document.getElementsByClassName('message')[0].textContent = serverResponse.userName + serverResponse.suffix;<br>&nbsp;&nbsp;}<br>};<br>const body = JSON.stringify({ userName: userName, suffix: ' loves cats!' });<br>xhr.send(body);</blockquote>
```
Comment (thecodingaviator) [inline curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/post-data-with-the-javascript-xmlhttprequest-method.english.md]: Could you please put each line of code on a different line, would improve readability

Comment (yoelvisj) [reply curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/post-data-with-the-javascript-xmlhttprequest-method.english.md]: The code maintains the line-by-line structure of the original challenge and affects the tests as little as possible. However, I look legible.

Comment (thecodingaviator) [reply curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/post-data-with-the-javascript-xmlhttprequest-method.english.md]: I see, it's fine

**Comment by ojeytonwilliams** [comment]
@HarplingeTom are you happy with these changes?

**Comment by thecodingaviator** [comment]
@HarplingeTom Just wanted a confirmation, you're fine with your code being merged via this PR. Right?

**Comment by jeremyfiel** [comment]
still pending for this challenge to update.  the changes reflected here work on the existing tests.  thanks!

**Comment by RandellDawson** [comment]
@QuincyLarson I saw you gave the OK to proceed on https://github.com/freeCodeCamp/freeCodeCamp/issues/14002.  I have validated this works, so if you are OK, we can go ahead and merge it.  The only reservations I have are that we are referring to `201` vs `200` response code and we are relying on a non-FCC resource link.  Is there any reason why we can not create a route on the FCC site like `requests/cat-lover` that would accomplish the same thing on something FCC controls?  Maybe for now, we go ahead and merge this one and then we can look at housing the url on the FCC side in a future PR.

File: curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/post-data-with-the-javascript-xmlhttprequest-method.english.md [hunk 2]
```diff
@@ -8,11 +8,11 @@ challengeType: 6
...
-Next, the <code>onreadystatechange</code> event listener handles a change in the state of the request. A <code>readyState</code> of 4 means the operation is complete, and a <code>status</code> of 200 means it was a successful request. The document's HTML can be updated.
-Finally, the <code>send</code> method sends the request with the <code>userName</code> value, which was given by the user in the <code>input</code> field.
+Next, the <code>onreadystatechange</code> event listener handles a change in the state of the request. A <code>readyState</code> of 4 means the operation is complete, and a <code>status</code> of 201 means it was a successful request. The document's HTML can be updated.
+Finally, the <code>send</code> method sends the request with the <code>body</code> value, which the <code>userName<code> key was given by the user in the <code>input</code> field.
```
Comment (RandellDawson) [inline curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/post-data-with-the-javascript-xmlhttprequest-method.english.md]: @yoelvisj Missing a `/` on the `code` tag after `userName`.
```suggestion
Finally, the <code>send</code> method sends the request with the <code>body</code> value, which the <code>userName</code> key was given by the user in the <code>input</code> field.
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
See my requested change.

**Review by moT01 — CHANGES_REQUESTED** [review]

**Review by RandellDawson — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
Looks good @yoelvisj - thanks for contributing to freeCodeCamp! 🎉

**Pull request merged**